# **Supermarket Sales Analysis: Binary Brogrammers Project**

### Project Overview
This supermarket sales analysis project is crucial to gaining a competitive advantage in today's dynamic retail landscape. Analyzing customer behaviour allows for informed decision-making, leading to **increased sales and profitability**. It needs to be more accurate in understanding buying patterns that could result in missed opportunities and loss of revenue. This project has the potential to provide insights that can be used to improve supermarket operations, enhance customer experience, and drive profitability.

### Business Problems
1. What are the most selling products and wich are the declining sales products?
2. Are there peak sales time (days, month, season)?
3. What is the most preferred payment method?
4. What shopping mall has the highest sales?
5. What Age group has the highest sales?

### Project Goals
1. Identify the most selling products and the declining sales products.
2. Identify peak sales time in terms of days, months, and seasons.
3. Determine the most preferred payment method.
4. Identify the shopping mall with the highest sales.
5. Identify the age group with the highest sales.

## Import Libraries

In [10]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Load Data

In [11]:
shopdat = pd.read_csv('../data/customer_shopping_data.csv')
print(f'Data Rows: {shopdat.shape[0]}\nData Columns: {shopdat.shape[1]}')

Data Rows: 99457
Data Columns: 10


## First look at the data

In [12]:
print(shopdat.info())
print(f'\nTotal Duplicated Rows:\n{shopdat.duplicated().sum()}')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99457 entries, 0 to 99456
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   invoice_no      99457 non-null  object 
 1   customer_id     99457 non-null  object 
 2   gender          99457 non-null  object 
 3   age             99457 non-null  int64  
 4   category        99457 non-null  object 
 5   quantity        99457 non-null  int64  
 6   price           99457 non-null  float64
 7   payment_method  99457 non-null  object 
 8   invoice_date    99457 non-null  object 
 9   shopping_mall   99457 non-null  object 
dtypes: float64(1), int64(2), object(7)
memory usage: 7.6+ MB
None

Total Duplicated Rows:
0


In [13]:
shopdat.head()

invoice_no customer_id  gender  age  category  quantity    price  \
0    I138884     C241288  Female   28  Clothing         5  1500.40   
1    I317333     C111565    Male   21     Shoes         3  1800.51   
2    I127801     C266599    Male   20  Clothing         1   300.08   
3    I173702     C988172  Female   66     Shoes         5  3000.85   
4    I337046     C189076  Female   53     Books         4    60.60   

  payment_method invoice_date   shopping_mall  
0    Credit Card     5/8/2022          Kanyon  
1     Debit Card   12/12/2021  Forum Istanbul  
2           Cash    9/11/2021       Metrocity  
3    Credit Card   16/05/2021    Metropol AVM  
4           Cash   24/10/2021          Kanyon

Summary findings from the first look of the dataset:
- Seems like `price` column is better renamed to total price because as we can see at the 3rd row, the price for 1 clothing is 300.08 while on the first row the clothing price for 5 quantity is 1500.40 which is the total price for 5 quantity. So, I will rename the column to total_price.
- We can extract `invoice_date` to get the year, month, day, weekend,seasons
- We can group `age` into age groups based on the `age` distribution itself
